src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -56,6 +59,16 @@ public HttpsConnectionMiddleware(ConnectionDelegate next, HttpsConnectionAdapter
                 }
```

Comment:
On Win8 for Http1AndHttp2 we downgrade you to Http1, but for Http2 we let it through? I think we should also be changing line 56 above to check for Win10 and updating the message to match.

This is the default for the gRPC template.

Reply:
Actually this part seems fishy. The logic throws below 6.2 but ALPN was added in win8.1 so should it throw on 6.2 as well? As for checking for Win10, we concluded it was possible to update the registry keys to make it work on 8.1 right? If that's the case, we shouldn't throw.

Reply:
Update behaviour, I'll check to make sure all tests are updated.